Assets/Scripts/Gameplay/EndPoint.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using UnityEngine;
+
+namespace Gameplay
+{
+    public class EndPoint : MonoBehaviour
+    {
+        public event Action LevelPassed;
+        
+        
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.GetComponent<Player>())
+            {
+                LevelPassed?.Invoke();
```

Comment:
Need Inject IGameRun to this class. And invoke finish

Reply:
Done